plugins/processors/awsappsignals/customconfiguration/replacer.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package customconfiguration
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type ReplaceActions struct {
+	Actions []ActionItem
+}
+
+func NewCustomReplacer(rules []Rule) *ReplaceActions {
+	return &ReplaceActions{
+		generateActionDetails(rules, "replace"),
+	}
+}
+
+func (r *ReplaceActions) Process(attributes, _ pcommon.Map, isTrace bool) error {
+	// do nothing when there is no replace rule defined
+	if r.Actions == nil || len(r.Actions) == 0 {
+		return nil
+	}
+	// If there are more than one rule are matched, the last one will be executed(Later one has higher priority)
+	actions := r.Actions
+	finalRules := make(map[string]string)
+	for i := len(actions) - 1; i >= 0; i = i - 1 {
+		element := actions[i]
+		isMatched, _ := isSelected(attributes, element.SelectorMatchers, isTrace)
+		if !isMatched {
+			continue
+		}
+		for _, replacement := range element.Replacements {
+			targetDimensionKey := getExactKey(replacement.TargetDimension, isTrace)
+			// don't allow customer add new dimension key
+			_, isExist := attributes.Get(targetDimensionKey)
+			if !isExist {
+				continue
+			}
+			// every replacement in one specific dimension only will be performed once
+			_, ok := finalRules[targetDimensionKey]
+			if ok {
+				continue
+			}
+			finalRules[targetDimensionKey] = replacement.Value
```

Comment:
nit: Why not just do the `attributes.PutStr` here? Instead of having the `finalRules` be a `map[string]string`, you can have it be a `collections.Set[string]` so it serves the same purpose.

Reply:
To me, it's more of a personal coding style choice. I actually prefer the original version cuz it would make the actual writing to original `attributes` happen in only one place.

Unless there's a strong reason to change it, I'll keep what it is. :)